docs/DafnyRef/Expressions.md

```diff
@@ -567,12 +567,19 @@ an instance method or function.
 ## 20.21. Fresh Expressions {#sec-fresh-expression}
 ````grammar
 FreshExpression_ =
-  "fresh" "(" Expression(allowLemma: true, allowLambda: true) ")"
+  "fresh" [ "@" LabelName ]
+  "(" Expression(allowLemma: true, allowLambda: true) ")"
 ````
 
 `fresh(e)` returns a boolean value that is true if
-the objects referenced in expression `e` were all
-freshly allocated in the current method invocation.
+the objects denoted by expression `e` were all
+freshly allocated since the time of entry to the enclosing method.
```

Comment:
`fresh` is a built-in two-state predicate, but its argument is like that of a user-defined two-state predicate. That is, the argument is one expression.

In contrast, `unchanged` is more like `modifies`. Semantically, it can be used to revert what a `modifies` clause says. As such, its arguments are like those of a `modifies` clause. This allows field specifications like `` `f``, which are only allowed in frame expressions. And since `modifies` can take a list of frame expressions, it seemed natural to let `unchanged` do the same.